File: binsV1/binV14/second_problem.js
const posts = {
  post_1: { tags: ["js", ["cli", "json"], "parsing"] },
  post_2: { tags: ["js", ["api", ["cli"]]] },
};

const tags = Object.values(posts).flatMap((post) => post.tags.flat(5));

const uniqueTags = [...new Set(tags)];

console.log(JSON.stringify({ data: uniqueTags }, null, 2));

File: binsV1/binv10/second_problem.rb
require 'json'

posts = {
  "post_1": { "tags": [ "ruby", [ "cli", "json" ], "parsing" ] },
  "post_2": { "tags": [ "ruby", [ "api", [ "cli" ] ] ] }
}

tags = posts.values.flat_map do |post|
  post[:tags].flatten
end.uniq

puts JSON.pretty_generate({ data: tags })
